test/e2e/llmisvc/diagnostic.py

```
@@ -0,0 +1,98 @@
+            src = f"{ev.source.component or ''}/{ev.source.host or ''}".strip("/")
+            msg = (ev.message or "").replace("\n", " ")
+            print(
+                f"{ts_str:<25} {ev.metadata.namespace:<12} {src:<20} {ev.type or '':<8} "
+                f"{ev.reason or '':<20} {msg}"
+            )
+
+    except Exception as e:
+        print(f"# ❌ failed to list events: {e}")
+
+
+def kinds_matching_by_labels(namespace: str, labels, skip_api_kinds={"Secret"}):
```

Ruff (0.12.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)